lib/Sema/MiscDiagnostics.cpp

```diff
@@ -1940,7 +1940,18 @@ class VarDeclUsageChecker : public ASTWalker {
     return (VarDecls[VD] & RK_Written) != 0;
   }
 
+  bool isImplicitNewValueInSetter(VarDecl *VD) {
+    auto FD = dyn_cast<FuncDecl>(VD->getDeclContext());
+
+    return (VD->isImplicit() && VD->getName().str() == "newValue" &&
```

Comment:
Rather than matching on the name string, check the appropriate parameter list of the enclosing function.